### PR TITLE
DOC: Improve consistency in random module docstrings

### PR DESCRIPTION
**Pull request by gmcastil**
This is a revision to an earlier PR (#5651) which I had mistakenly merged something else into.  I reverted return value descriptions - these still need some standardization.  I also fixed a number of broken links and references.  I'll try to suggest a cleaner return value description, since so many of them are similar.

### CODE REVIEW

**Comment by gmcastil** [comment]
No sure why Travis-CI failed, it passed all the tests on my end.

**Comment by jaimefrio** [comment]
I have restarted that build for everyone's peace of mind. But it looks like an issue on Travis setting up the environment, you needn't worry much.

**Comment by charris** [comment]
The `ftp.waf.io` site is offline. I wonder if that might be related to google code shutdown? There is a backup site, but we might want to remove the bento check for a while.

**Comment by gdementen** [comment]
One nitpick: you introduced a few instances of `"abcd,"` (ie a coma at the end of a paper title), but I suppose it should read `"abcd",` instead.

**Comment by gmcastil** [comment]
@gdementen I know.  The example given in section 11 of this[https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt] showed the comma on the inside.  But as I was going through it, I realized that the module seemed to be more or less consistently putting it on the outside.  I'll do a separate PR just covering references and make it consistent throughout the module.

**Comment by charris** [comment]
Rejigger tests by closing and opening.

File: numpy/random/mtrand/mtrand.pyx [hunk 1]
```diff
@@ -2373,15 +2382,15 @@ cdef class RandomState:
...
+        The t test is based on an assumption that the data come from a
+        Normal distribution. The t test provides a way to test whether
+        the sample mean (that is the mean calculated from the data) is
+        a good estimate of the true mean.
 
-        The derivation of the t-distribution was forst published in 1908 by William
-        Gisset while working for the Guinness Brewery in Dublin. Due to proprietary
-        issues, he had to publish under a pseudonym, and so he used the name
-        Student.
+        The derivation of the t-distribution was forst published in
```
Comment (charris) [inline numpy/random/mtrand/mtrand.pyx]: forst -> first.

File: numpy/random/mtrand/mtrand.pyx [hunk 2]
```diff
@@ -2885,28 +2911,27 @@ cdef class RandomState:
...
+        than the standard Gaussian distribution.
 
         References
         ----------
-        .. [1] Abramowitz, M. and Stegun, I. A. (Eds.). Handbook of Mathematical
-               Functions with Formulas, Graphs, and Mathematical Tables, 9th
-               printing.  New York: Dover, 1972.
+        .. [1] Abramowitz, M. and Stegun, I. A. (Eds.). "Handbook of
+               Mathematical Functions with Formulas, Graphs, and Mathematical
+               Tables, 9th printing," New York: Dover, 1972.
```
Comment (charris) [inline numpy/random/mtrand/mtrand.pyx]: Looks like most of the reference sections don't have blank lines between the entries.

File: numpy/random/mtrand/mtrand.pyx [hunk 3]
```diff
@@ -2974,59 +2999,55 @@ cdef class RandomState:
...
-        :math:`\\frac{\\pi^2}{6}\\beta^2`.
+        The function has a mean of :math:`\\mu + 0.57721\\beta` and a variance
+        of :math:`\\frac{\\pi^2}{6}\\beta^2`.
 
         References
         ----------
-        Gumbel, E. J., *Statistics of Extremes*, New York: Columbia University
-        Press, 1958.
+        .. [1] Gumbel, E. J., "Statistics of Extremes,"
+               New York: Columbia University Press, 1958.
```
Comment (charris) [inline numpy/random/mtrand/mtrand.pyx]: Blank line again.

File: numpy/random/mtrand/mtrand.pyx [hunk 4]
```diff
@@ -3885,8 +3916,9 @@ cdef class RandomState:
 
         References
         ----------
-        Zipf, G. K., *Selected Studies of the Principle of Relative Frequency
-        in Language*, Cambridge, MA: Harvard Univ. Press, 1932.
+        .. [1 ]Zipf, G. K., "Selected Studies of the Principle of Relative
```
Comment (charris) [inline numpy/random/mtrand/mtrand.pyx]: `[1] `.

**Comment by charris** [comment]
Very nice clean up, just a couple of nitpicks. It would be good to run a spell check.

**Comment by charris** [comment]
Did some additional cleanups and merged in 4679ff2..061aa26. Thanks @gmcastil.

**Pull request closed without merging**